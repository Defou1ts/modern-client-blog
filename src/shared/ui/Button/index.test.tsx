import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import { Button } from './index';

describe('Button', () => {
	it('renders a primary button', () => {
		render(<Button type="primary">Hello</Button>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('renders a secondary button', () => {
		render(<Button type="secondary">Hello</Button>);
		const text = screen.getByText('Hello');
		expect(text).toBeInTheDocument();
		expect(text).toBeVisible();
	});

	it('button primary snapshot', () => {
		const component = renderer.create(<Button type="primary">Hello</Button>);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('button secondary snapshot', () => {
		const component = renderer.create(<Button type="secondary">Hello</Button>);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
