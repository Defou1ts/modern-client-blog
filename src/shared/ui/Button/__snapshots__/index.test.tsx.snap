// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button button primary snapshot 1`] = `
<button
  className="button primary"
>
  Hello
</button>
`;

exports[`Button button secondary snapshot 1`] = `
<button
  className="button secondary"
>
  Hello
</button>
`;
